js/runner.go

```diff
@@ -318,9 +323,59 @@ func (r *Runner) SetOptions(opts lib.Options) error {
 		r.console = c
 	}
 
+	// FIXME: Resolver probably shouldn't be reset here...
+	// It's done because the js.Runner is created before the full
+	// configuration has been processed, at which point we don't have
+	// access to the DNSConfig, and need to wait for this SetOptions
+	// call that happens after all config has been assembled.
+	// We could make DNSConfig part of RuntimeOptions, but that seems
+	// conceptually wrong since the JS runtime doesn't care about it
+	// (it needs the actual resolver, not the config), and it would
+	// require an additional field on Bundle to pass the config through,
+	// which is arguably worse than this.
+	ttl, err := parseTTL(opts.DNS.TTL.String)
+	if err != nil {
+		return err
+	}
+	dnsSel := opts.DNS.Select.DNSSelect
+	if !dnsSel.IsADNSSelect() {
+		dnsSel = lib.DefaultDNSConfig().Select.DNSSelect
+	}
+	dnsPol := opts.DNS.Policy.DNSPolicy
+	if !dnsPol.IsADNSPolicy() {
+		dnsPol = lib.DefaultDNSConfig().Policy.DNSPolicy
+	}
+	r.Resolver = netext.NewResolver(r.ActualResolver, ttl, dnsSel, dnsPol)
+
 	return nil
 }
 
+func parseTTL(ttlS string) (time.Duration, error) {
+	ttl := time.Duration(0)
+	switch ttlS {
+	case "inf":
+		// cache "infinitely"
+		ttl = time.Hour * 24 * 365
+	case "0":
+		// disable cache
+	case "":
+		ttlS = lib.DefaultDNSConfig().TTL.String
+		fallthrough
+	default:
+		origTTLs := ttlS
+		// Treat unitless values as milliseconds
+		if t, err := strconv.ParseFloat(ttlS, 32); err == nil {
+			ttlS = fmt.Sprintf("%.2fms", t)
+		}
+		var err error
+		ttl, err = types.ParseExtendedDuration(ttlS)
+		if ttl < 0 || err != nil {
+			return ttl, fmt.Errorf("invalid DNS TTL: %s", origTTLs)
+		}
+	}
```

Comment:
Can we get this as a separate function in `lib.types` ? so we can use it wherever we need number to duration functionality (where the duration is ms if no unit is given)

Reply:
Instead of adding a new function, I added the ms logic to `types.ParseExtendedDuration` in 2dc9f1af, which seems to work fine.